src/components/Notifications/EventLog/EventLogTreeFilter.tsx

```diff
@@ -0,0 +1,207 @@
+                        const treeNode = prev[activeFilter.bundleId];
+                        const activeChips = activeFilter.chips.map(chip => chip.value);
+
+                        treeNode.children?.forEach(childNode => {
+                            childNode.checkProps.checked = activeChips.includes(childNode.id);
+                        });
+
+                        if (allChildrenChecked(treeNode)) {
+                            treeNode.checkProps.checked = true;
+                        } else if (childChecked(treeNode)) {
+                            treeNode.checkProps.checked = null;
+                        } else {
+                            treeNode.checkProps.checked = false;
+                        }
+                    });
+                }
+            }));
+        }
+    }, [ groups, filters, initialize ]);
+
+    const flattenTree = React.useMemo(() => {
+        const flatTreeDataArray = produce(treeDataArray, (prev) => {
+            prev.forEach(treeNode => {
+                if (treeNode.children) {
+                    prev.push(...treeNode.children);
+                }
+            });
+        });
+
+        return flatTreeDataArray;
+    }, [ treeDataArray ]);
+
+    React.useEffect(() => {
+        const groupNames = groups.map(group => group.name);
+        const activeParentFilters = flattenTree.filter(treeNode => {
+            const isActive = treeNode.checkProps.checked;
+            return (isActive || isActive === null) && groupNames.includes(treeNode.id);
+        });
+
+        updateFilters(produce((prev) => {
+            if (prev.length === 0 && activeParentFilters.length === 0) {
+                return prev;
+            } else if (prev.length !== 0 && activeParentFilters.length === 0) {
+                return [];
+            } else {
+                const newCustomFilters = activeParentFilters.map(parentFilter => ({
+                    bundleId: parentFilter.id,
+                    category: parentFilter.name as string,
+                    chips: parentFilter.children?.filter(childNode => childNode.checkProps.checked).map(childFilter => ({
+                        name: childFilter.name as string,
+                        value: childFilter.id,
+                        isRead: true
+                    }))
+                ??
+                    [
+                        {
+                            name: parentFilter.name,
+                            value: parentFilter.id,
+                            isRead: true
+                        }
+                    ]
+                } as EventLogCustomFilter));
+
+                const prevBundles = prev.map(prevFilter => prevFilter.bundleId);
+                const currBundles = newCustomFilters.map(customFilter => customFilter.bundleId);
+
+                const areBundlesEqual = areEqual(prevBundles, currBundles);
+                const areApplicationsEqual = newCustomFilters.every((entry, idx) => {
```

Comment:
add `areBundlesEqual` to this line to prevent unnecessary iterations if we know the filters are not equal